src/main/java/io/dataspaceconnector/common/net/ApiReferenceHelper.java

```diff
@@ -39,16 +47,20 @@ public final class ApiReferenceHelper {
     private String defaultBaseUrl;
 
     /**
-     * Checks whether a URL is a route reference by checking whether the URL starts with the path
-     * to this connector's routes API. The full path to the routes API consists of the application's
-     * base URL and the path segment for the routes API.
+     * Route repository used to check if a given url is a route.
+     */
+    private final RouteRepository routeRepository;
+
+    /**
+     * Checks whether a URL is a route reference by searching for
+     * the uuid of the url in the route repository.
      *
      * @param url The URL to check.
      * @return True, if the URL is a route reference; false otherwise.
      */
     public boolean isRouteReference(final URL url) {
-        final var routesApiUrl = getBaseUrl() + BasePath.ROUTES;
-        return url.toString().startsWith(routesApiUrl);
+        final var id = url.getPath().substring(url.getPath().lastIndexOf('/') + 1);
```

Comment:
I would prefer to go with a regex, which searches for the UUID at the end of the string: https://stackoverflow.com/questions/136505/searching-for-uuids-in-text-with-regex. Something like `.*\\[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$` may work